src/dsp_tools/utils/xmlupload/read_validate_xml_file.py

```diff
@@ -9,50 +8,39 @@
 from dsp_tools.models.exceptions import UserError
 from dsp_tools.models.xmlresource import XMLResource
 from dsp_tools.utils.create_logger import get_logger
+from dsp_tools.utils.shared import validate_xml_against_schema
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
 
 logger = get_logger(__name__)
 
 
-def parse_xml_file(input_file: Union[str, Path, etree._ElementTree[Any]]) -> etree._Element:
+def validate_and_parse_xml_file(
```

Comment:
I'm a bit unhappy with the order: The correct order would be: First parse, then validate. In the code, down on lines 36-37, the order is also wrong.

When looking into it, I realized that you cannot simply inverse lines 36-37, because you inherited a mess from me: there are a bunch of functions, but the way they are organized is not intuitive.

I see a way how I could improve it, but I would need to do it myself (or in pair programming with you). It's too complicated to explain :-) 

In fact I would like to improve it now, because there is no worth splitting long functions into smaller ones, if the "Zuständigkeiten" of them are not clean. What do you think?

Reply:
I agree with the order, it is not intuitive. But the only thing I did was use the extract method from pycharm refactor (smart copy-paste) and moved the functions around. I would prefer to close this down quickly. So it would be fine for me if you did it later.